scenarios/turbine-heat-sensor/src/main/java/io/pravega/turbineheatsensor/TurbineHeatSensor.java

```diff
@@ -142,6 +142,7 @@ public static void main(String[] args) throws Exception {
         if ( !onlyWrite ) {
             consumeStats.printTotal();
         }
+        clientFactory.close();
```

Comment:
This is a bit confusing as we are closing in the `main` method and in the `runLoop`. Is the contract that the runnable is supposed to close the factory or is this something that needs to be done by the caller? Let's be consistent about how we are closing it.

Reply:
To follow a consistent pattern without changing much the current code: i) now there is only one client factory that is created and closed within the main method. ii) Writers and readers take as input that factory to instantiate an EventStream[Writer/Reader], iii) writers and readers are in charge of closing their own EventStream[Writer/Reader]. In any case, a future PR for issue #119 will be devoted to improve the code of this application, so we can find better ways of managing resources (out of the scope of this PR).